### PR TITLE
Dotcom Paid Stats: Fixes for Odyssey on Simple and Atomic

### PR DESCRIPTION
**Pull request by kangzj**
Related to #

## Proposed Changes

* Add `is_odyssey` flag, which identifies Stats is running in WP Admin as `is_running_in_jetpack_site` return false for Simple sites yet it is still running Odyssey Stats. 
* `is_running_in_jetpack_site` now only means Stats calls public-api directly
* Fix purchase link and support links

## Why are these changes being made?

* Ensure upsell works for Odyssey sites

### Simple

* Open your local Calypso dev env
* Run `NODE_ENV=production yarn dev --sync` to sync to your sandbox
* Sandbox `widgets.wp.com`
* Open a Free Simple site, which has Classic admin interface (You can enable it in general settings)
* Open `/wp-admin/admin.php?page=stats#!/stats/day/:siteSlug?flags=stats%2Fpaid-wpcom-v3`
* Ensure you see the v3 upsell
* Click Insight page
* Ensure you see the v3 upsell
* Ensure both upsell directs you to checkout
* Checkout personal
* Ensure upsell for Personal goes away
* Ensure you can upgrade to Premium
* Ensure no blocked modules after upgrading to Premium

image

### WoA

* Open an Atomic site `/wp-admin/admin.php?page=stats#!/stats/day/:siteSlug?flags=stats%2Fpaid-wpcom-v3`
* Ensure nothing is paywalled

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by kangzj** [comment]
Inline support link is something we'll look into later. We might or might not be able to implement it for Odyssey Stats.

[Tracked](https://github.com/Automattic/red-team/issues/313)

File: client/my-sites/stats/stats-upsell/index.tsx [hunk 1]
```diff
@@ -53,6 +53,7 @@ export default function StatsUpsell( { title, features, image, statType }: Props
 		recordTracksEvent( `${ eventPrefix }_stats_upsell_submit`, {
 			stat_type: statType,
 		} );
+		// TODO: add redirect_to param so that we can redirect back to the stats page after checkout.
```
Comment (kangzj) [inline client/my-sites/stats/stats-upsell/index.tsx]:
```suggestion
```

**Review by lsl — APPROVED** [review]
Learn more and checkout buttons now working.

> Ensure upsell for Personal goes away

This didn't happen for me, but I believe it was some kind of caching issue. We should keep an eye out for user reports about this, there might be some invalidation we can/should be doing. I did not see the same thing happen when upgrading to premium, just personal.

Tested with and without the diff, everything looks working as intended.

File: apps/odyssey-stats/src/lib/create-odyssey-config.ts [hunk 2]
```diff
@@ -119,6 +119,11 @@ export class ConfigApi extends Function {
 		// Note: configData is hydrated in https://github.com/Automattic/jetpack/blob/d4d0f987cbf63a864b03b542b7813aabe87e0ed3/projects/packages/stats-admin/src/class-dashboard.php#L214
 		this.configData.features = productionConfig.features;
 
+		// Set a flag to identify if the app is running in WP Admin.
+		// `is_running_in_jetpack_site` now means whether the app calls public-api directly or use the Jetpack ones.
+		// For Simple sites running Odyssey Stats, `is_running_in_jetpack_site` is true and `is_odyssey` is false.
```
Comment (dognose24) [inline apps/odyssey-stats/src/lib/create-odyssey-config.ts:124]:
```suggestion
        // For Simple sites running Odyssey Stats, `is_running_in_jetpack_site` is false and `is_odyssey` is true.
```
Is the comment correct with this change?

**Pull request merged**